habitat_baselines/common/base_trainer.py

```diff
@@ -0,0 +1,48 @@
+from abc import ABC, abstractmethod
+
+from habitat_baselines.common.trainer_registry import trainer_registry
+
+
+class BaseTrainer(ABC):
+    def __init__(self):
+        pass
+
+    @abstractmethod
+    def train(self):
+        pass
+
+    @abstractmethod
+    def eval(self):
+        pass
+
+    @abstractmethod
+    def save_checkpoint(self, file_name):
+        pass
+
+    @abstractmethod
+    def load_checkpoint(self):
+        pass
+
+
+def get_trainer(trainer_name, trainer_cfg):
+    trainer = trainer_registry.get_trainer(trainer_name)
+    assert trainer is not None, f"{trainer_name} is not supported"
+    return trainer(trainer_cfg)
+
+
+class BaseRLTrainer(BaseTrainer):
```

Comment:
Is there a point in having this when it is not adding any new functionality on top of `BaseTrainer`?

Reply:
@apsdehal The rationale is that `common/base_trainer.py` should in the future support imitation learning and SLAM based agents and `BaseTrainer` would be a template for those. I do think we need more discussion on what functionalities to be included inside `BaseTrainer` vs. `BaseRLTrainer`